frontend/src/constants/environments.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Define environment variables that you need exposed in the client-side bundle.
+ * These should not include sensitive secrets!
+ */
+const PUBLIC_ENV_VARS_BY_ENV = {
+  development: {
+    GOOGLE_ANALYTICS_ID: "GTM-MV57HMHS",
+    GTM_AUTH: "Xf-LmsD6dhZRJXZz21rZVA",
+    GTM_PREVIEW: "env-8",
+  },
+  test: {
+    GOOGLE_ANALYTICS_ID: "GTM-MV57HMHS",
+    GTM_AUTH: "73_zp32T8ExOVz-f_X56dQ",
+    GTM_PREVIEW: "env-9",
+  },
+  production: {
+    GOOGLE_ANALYTICS_ID: "GTM-MV57HMHS",
+    GTM_AUTH: "hqnD044nMRMTQ0C3XBpbfQ",
+    GTM_PREVIEW: "env-1",
+  },
+} as const;
+
+const NEXT_ENVS = ["development", "test", "production"] as const;
+export type NextPublicAppEnv = (typeof NEXT_ENVS)[number];
+
+const CURRENT_ENV = process.env.NODE_ENV ?? "development";
```

Comment:
Where is `NODE_ENV` being set? Our application is environment agnostic at the moment, and prefer it that way (have intentionally made decisions to maintain that). Just not sure this will hit anything but `"development"` at this point.

Reply:
Hmm, I see one point where it's being set here:

`reloadOnPrerender: process.env.NODE_ENV === "development",`

Reply:
Thanks for explaining offline it's the different way we start the app.

Reply:
`NODE_ENV` is set when we run `npm run start` automagically by node to `development` and that is a flag that node uses to set certain configs. `npm run test` sets it to `test`, and serving it from the `server.js` file sets it to `production`. We could have more finely tuned control by using env files, but that would add some overhead that I don't think we need yet.